### ISSUE TITLE
Implement UDP/IP socket function to send and receive camera data

### ISSUE THREAD

**Issue by Talkasi** [Talkasi/call-app#6]
In this issue should be first found out if it is better to use one UDP/IP socket or two. **Camera function can be changed in any way is needed.**

If it is better to use one socket, camera data should be sent from server to the client and from client to the server.

Both server and client should be able to:
- get data from camera and send it;
- recieve camera data and display it.

If it is better to use two sockets, then we actually need only two functions: one for client part and one for server part. Then we can simply run server and client parts of two sockets on one machine. Client should get camera data and send it to server. Server should recieve this data and display it.

**Comment by Talkasi** [comment]
It is possible to send and receive data by both server and client [like that](https://www.geeksforgeeks.org/udp-server-client-implementation-c/). But we also need our call app to work without lags, in this solution we must wait until one pack of data would be received and only then send another pack of data.  Maybe it is more convenient to use two sockets instead of one: if we have enough resources, we can just thread them and it will work better, because we will not wait too long. But if it will not be so long, we can use one socket, I prefer this variant.

@anton2920, what do you think?

**Comment by anton2920** [comment]
> It is possible to send and recieve data by both server and client [like that](https://www.geeksforgeeks.org/udp-server-client-implementation-c/).

What seems impossible about this? :)

> But we also need our call app to work without lags, in this sollution we must wait untill one pack of data would be received and only then send another pack of data.

Why must we wait? Isn't the whole point of lossy connection with UDP _not_ to wait for anything and just «spam» the data as soon as it's there?

>  Maybe it is more convinient to use two sockets instead of one: if we have enough resorses, we can just thread them and it will work better, because we will not wait too long. But if it will not be so long, we can use one socket, I prefer this variant.

I think you can still have threading just with one socket.

**Comment by Talkasi** [comment]
> Why must we wait? Isn't the whole point of lossy connection with UDP not to wait for anything and just «spam» the data as soon as it's there?

Yes, it is exactly what I want to implement, but we can't send the data before we are not done with receiving another pack of data.

**Comment by anton2920** [comment]
> Yes, it is exactly what I want to implement, but we can't send the data before we are not done with receiving another pack of data.

I mean, why? Is there a technical limitation for that?

**Comment by Talkasi** [comment]
@anton2920, please take a look.

**Comment by anton2920** [comment]
@Talkasi, please see discussions.
